### PR TITLE
Lazarus: Add version 2.0.0

### LINKED ISSUE #1577: [Request] Add Lazarus pascal compiler
Pascal compiler

* lazarus ide https://www.lazarus-ide.org/
* choco package https://chocolatey.org/packages/lazarus
* package source https://github.com/MicksMix/chocolatey-lazarus

Please and thank you!

### PR DESCRIPTION
**Pull request by niheaven**
Close #1577

### CODE REVIEW

File: bucket/lazarus.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            "hash": "e53f18e15ff94998cda968325214d11326f20f148b47f349230151dbfc11ca70"
+        },
+        "32bit": {
+            "url": "https://downloadssourceforge.net/project/lazarus/Lazarus%20Windows%2032%20bits/Lazarus%202.0.0/lazarus-2.0.0-fpc-3.0.4-win32.exe",
+            "hash": "fb64be2210c3583f3a184a4a3acc2342624582f7d946ab06d4faf64c81e0326d"
+        }
+    },
+    "innosetup": true,
+    "post_install": [
+        "$fpcdir = \"$dir\\fpc\\\" + ([Regex]'fpc-([\\d.]+)').Matches($fname).Groups[1].Value",
```
Comment (Ash258) [inline bucket/lazarus.json]: Is it needed to call regex library? Something like `-replace '..', '$1' should work too.
```suggestion
        "$fpcdir = \"$dir\\fpc\\\" + ("$fname" -replace 'fpc-([\\d\\.]+)', '$1')
```

File: bucket/lazarus.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
+{
+    "homepage": "https://www.lazarus-ide.org",
+    "version": "2.0.0",
+    "description": "A Delphi compatible cross-platform IDE for Rapid Application Development.",
+    "license": {
+        "identifier": "GPL-2.0-or-later | FPC-modified-LGPL-2.0-or-later",
+        "url": "https://spdx.org/licenses/GPL-2.0-or-later.html; http://wiki.freepascal.org/FPC_modified_LGPL"
```
Comment (r15ch13) [inline bucket/lazarus.json]:
```suggestion
        "url": "http://wiki.lazarus.freepascal.org/licensing"
```

File: bucket/lazarus.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            "lazarus.exe",
+            "Lazarus"
+        ],
+        [
+            "startlazarus.exe",
+            "Lazarus (debug)",
+            "--debug"
+        ]
+    ],
+    "checkver": "lazarus-([0-9.]+)-fpc-(?<fpc>[0-9.]+)-win64.exe",
```
Comment (r15ch13) [inline bucket/lazarus.json]:
```suggestion
    "checkver": "lazarus-([\\d.]+)-fpc-(?<fpc>[\\d.]+)-win64.exe",
```

Comment (niheaven) [reply bucket/lazarus.json]: Are these two expressions equivalent?

File: bucket/lazarus.json (hunk 1 above)
Comment (niheaven) [reply bucket/lazarus.json]: Thanks, `-replace` does works.

File: bucket/lazarus.json (hunk 3 above)
Comment (r15ch13) [reply bucket/lazarus.json]: They are the same, but `\d` is almost consistently used in manifests :smile:

**Pull request merged**